Sources/SwiftCrossUI/Views/Shapes/Shape.swift

```diff
@@ -0,0 +1,125 @@
+/// A 2-D shape that can be drawn as a view.
+///
+/// If no stroke color or fill color is specified, the default is no stroke and a fill of the
+/// current foreground color.
+public protocol Shape: View
+where Content == EmptyView {
+    /// Draw the path for this shape.
+    ///
+    /// The bounds passed to a shape that is immediately drawn as a view will always have an
+    /// origin of (0, 0). However, you may pass a different bounding box to subpaths. For example,
+    /// this code draws a rectangle in the left half of the bounds and an ellipse in the right half:
+    /// ```swift
+    /// func path(in bounds: Path.Rect) -> Path {
+    ///     Path()
+    ///         .addSubpath(
+    ///             Rectangle().path(
+    ///                 in: Path.Rect(
+    ///                     x: bounds.x,
+    ///                     y: bounds.y,
+    ///                     width: bounds.width / 2.0,
+    ///                     height: bounds.height
+    ///                 )
+    ///             )
+    ///         )
+    ///         .addSubpath(
+    ///             Ellipse().path(
+    ///                 in: Path.Rect(
+    ///                     x: bounds.center.x,
+    ///                     y: bounds.y,
+    ///                     width: bounds.width / 2.0,
+    ///                     height: bounds.height
+    ///                 )
+    ///             )
+    ///         )
+    /// }
+    /// ```
+    func path(in bounds: Path.Rect) -> Path
+    /// Determine the ideal size of this shape given the proposed bounds.
+    ///
+    /// The default implementation accepts the proposal and imposes no practical limit on
+    /// the shape's size.
+    /// - Returns: Information about the shape's size. The ``ViewSize/size`` property is what
+    ///   frame the shape will actually be rendered with if the current layout pass is not
+    ///   a dry run, while the other properties are used to inform the layout engine how big
+    ///   or small the shape can be. The ``ViewSize/idealSize`` property should not vary with
+    ///   the `proposal`, and should only depend on the shape's contents. Pass `nil` for the
+    ///   maximum width/height if the shape has no maximum size (and therefore may occupy
```

Comment:
Remove the comment comment about occupying the entire screen because I can see it being a bit misleading; a shape with unlimited size may still occupy less than the entire screen if it likes to keep its aspect ratio etc.

Reply:
done